yt_dlp/extractor/youtube/_base.py

```diff
@@ -595,6 +601,31 @@ def _get_sid_authorization_header(self, origin='https://www.youtube.com', user_s
 
         return ' '.join(authorizations)
 
+    @property
+    def is_authenticated(self):
+        return self._has_auth_cookies
+
+    @property
+    def _has_auth_cookies(self):
+        yt_sapisid, yt_1psapisid, yt_3psapisid = self._get_sid_cookies()
+        # YouTube doesn't appear to clear 3PSAPISID when rotating cookies (as of 2025-04-26)
+        # But LOGIN_INFO is cleared and should exist if logged in
+        has_login_info = 'LOGIN_INFO' in self._youtube_cookies
+        return has_login_info and (yt_sapisid or yt_1psapisid or yt_3psapisid)
```

Comment:
this? since otherwise we could return the value of `yt_sapisid` etc
```suggestion
        return bool(has_login_info and (yt_sapisid or yt_1psapisid or yt_3psapisid))
```